src/main/java/at/hannibal2/skyhanni/features/mining/PowderPerHotmPerk.kt

```diff
@@ -0,0 +1,65 @@
+package at.hannibal2.skyhanni.features.mining
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.HotmData
+import at.hannibal2.skyhanni.events.LorenzToolTipEvent
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.round
+import at.hannibal2.skyhanni.utils.NumberUtil.addSeparators
+import at.hannibal2.skyhanni.utils.StringUtils.removeColor
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class PowderPerHotmPerk {
+
+    private val config get() = SkyHanniMod.feature.mining.hotmConfig
+
+    @SubscribeEvent
+    fun onTooltip(event: LorenzToolTipEvent) {
+        if (!isEnabled()) return
+
+        val itemName = event.itemStack.displayName
+        val perkEnum = HotmData.getPerkByNameOrNull(itemName.removeColor()) ?: return
+
+        if (perkEnum.getLevelUpCost() == null) return
+
+        val currentPowderSpend = perkEnum.calculateTotalCost(perkEnum.activeLevel)
+        val maxPowderNeeded = perkEnum.calculateTotalCost(perkEnum.maxLevel)
```

Comment:
Save the max Powder in the enum. Since it is literally an constant, so no need to calculate it every render frame.